docs/components/DataList/DataList.stories.mdx

```diff
@@ -23,3 +25,201 @@ item.
 If you need to add more information, consider using the
 [DataTable](?path=/story/components-lists-and-tables-datatable-docs--page)
 component instead.
+
+## Implementation guidelines
+
+Since DataList is a large component, we use a mixture of compound components and
+props to separate each part of the component.
+
+## Main wrapper
+
+DataList is the main component that holds everything and important props.
+
+<ArgsTable of={DataList} />
+
+## Layout
+
+DataList.Layout enables you to implement the layout of the list items. Keep in
+mind that DataList doesn't have a default layout so you have to have at least 1
+implemented.
+
+It also supports different `size` breakpoints so you can completely change the
+component it's using depending on the window size.
+
+It is also important to add the type of `DataListItem<typeof data>` on the
+`item` param in the layout so you can easily access the known data.
+
+```tsx
+// You can have 1 responsive layout
+<DataList>
+  <DataList.Layout>
+    {/* Declare the type of the item */}
+    {(item: DataListItem<MyDataType>) => (
+      <Grid>
+        <Grid.Cell size={{ xs: 12 md: 6 }}>{item.name}</Grid.Cell>
+        <Grid.Cell size={{ xs: 12 md: 6 }}>{item.email}</Grid.Cell>
+      </Grid>
+    )}
+  </DataList.Layout>
+</DataList>
+
+// Or you can have multiple layouts per breakpoint
+<DataList>
+  <DataList.Layout size="xs">
+    {(item: DataListItem<MyDataType>) => (
+      <Grid>
+        <Grid.Cell size={{ md: 6 }}>{item.name}</Grid.Cell>
+        <Grid.Cell size={{ md: 6 }}>{item.email}</Grid.Cell>
+      </Grid>
+    )}
+  </DataList.Layout>
+
+  <DataList.Layout size="md">
+    {(item: DataListItem<MyDataType>) => (
+      <Content>
+        <div>{item.name}</div>
+        <div>{item.email}</div>
+      </Content>
+    )}
+  </DataList.Layout>
+</DataList>
+```
+
+<ArgsTable of={DataList.Layout} />
+
+## Per item action(s)
+
+DataList.ItemActions allows you to add an action on the item itself such as
+`onClick` or `url`. While the DataList.ItemAction (without an s) child adds more
```

Comment:
I don't fully understand this guidance

Reply:
So `ItemActions` is a wrapper for `ItemAction`?

Reply:
Yup and also item actions also have props that enables the row click to navigate, etc.

I did struggle to write this one haha

Reply:
Even looking at the example it's a little hard to follow

It looks like `ItemActions` handles the entire row action, but it's just one action, so the plural name is confusing.

And then it wraps individual `ItemAction` actions

So I would expect a relationship like BottomSheet/BottomSheetOption, but it feels like they're actually more decoupled than that?

Reply:
yeah, a bit more decouple in a sense that you're just telling the data list that here are the list of actions you could do in this item. Also you can fire this onClick if you click the item itself.

I see what you're saying tho that maybe the row click should be its own compound

Reply:
OK so maybe this is a clearer way to break it down...

> DataList.ItemActions defines the action that will occur when the user clicks on the list item.
> 
> Provide a DataList.ItemAction for each additional action that you want to add. These additional actions should be presented in a context menu.

Or something like that?